app/models/service_order.rb

```diff
@@ -0,0 +1,15 @@
+class ServiceOrder < ActiveRecord::Base
+  belongs_to :tenant
+  belongs_to :user
+  has_many :miq_requests, :dependent => :nullify, :after_add => :assign_user_name_to
+
+  validates :state, :inclusion => {:in => %w(wish cart ordered)}
+  validates :name, :presence => true
+  validates :state, :presence => true
+
+  default_value_for :user_id, User.current_user.id
+
+  def assign_user_name_to(miq_request)
+    miq_request.user_name = user.name
```

Comment:
Since we want to capture the user_name of the user the service order is for, we should probably do something like `self.user_name = User.current_user.id` in a before create and not in default_value_for.